work16/index.js

```diff
@@ -0,0 +1,248 @@
+
+const REQUEST_URL = "https://mocki.io/v1/21ee587f-5a59-40ff-98c6-46b2d4715f25"
+
+// 間違っているURLの場合↓
+// const REQUEST_URL = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// 空の配列の場合↓
+// const REQUEST_URL = "https://mocki.io/v1/242b685f-a3d7-45a8-aeca-0376bd495b89";
+// 503エラーの場合↓
+// const REQUEST_URL = "https://httpstat.us/503";
+
+const request = async () => {
+    const response = await fetch(REQUEST_URL, {
+        headers: {
+            "Content-Type": "application/json"
+        }
+    });
+    if (!response.ok) {
+        console.log(`${response.status}:${response.statusText}`);
+    } else {
+        return response.json();
+    }
+}
+
+const init = async () => {
+    renderLoadingImg();
+
+    let responseJsonData;
+
+    try {
+        responseJsonData = await request();
+
+        if (!responseJsonData) {
+            return;
+        }
+    } catch (error) {
+        displayErrorMessage(error.message);
+    }
+    finally {
+        hideLoadingImg();
+    }
+
+    if (!responseJsonData.data.length) {
+        displayErrorMessage("data is empty");
+        return;
+    }
+    displayNews(responseJsonData.data);
+}
+
+const displayErrorMessage = (error) => {
+    const p = document.createElement("p");
+    p.textContent = error;
+    document.body.prepend(p);
+}
+
+const displayNews = (newsDataArray) => {
+    renderNewsTab(newsDataArray);
+    renderNewsContent(newsDataArray);
+}
+
+const renderNewsTab = (newsDataArray) => {
+    const tabList = document.getElementById("js_tabList");
+    const fragment = document.createDocumentFragment();
+
+    newsDataArray.forEach((newsCategoryObj, newsCategoryIndex) => {
+        const li = document.createElement("li");
+        const a = document.createElement("a");
+
+        if (!newsCategoryIndex) {
+            li.setAttribute("aria-selected", true);
+        } else {
+            li.setAttribute("aria-selected", false);
+        }
```

Comment:
MUST
Which tab is initially selected is included in the data.
> どのカテゴリタブを初期表示時に選んでいるかはデータとして持っている

Reply:
Sorry.
I missed it completely.
I have corrected it here.
Please check it.

commit: [a8a25fc](https://github.com/yasu-pro/js-handson/pull/18/commits/a8a25fc20d1230d115cb03f8ed233dd26afe620c)
commit: [41c9d42](https://github.com/yasu-pro/js-handson/pull/18/commits/41c9d422eeee9f0ee224c8b1200ec3c042c813b5)